### ISSUE TITLE
Update contribution guide

### ISSUE THREAD

**Issue by daxpryce** [microsoft/graspologic#503]
The contribution guide has a ton of good information in it, but we want to take it one step further and really focus on a few more things we want to get from a good PR.

One of the challenges in writing mathematic/scientific libraries is that it's often written as if the consuming user is a mathematician themselves.  One thing we aim to do with this project is bridge the gap between the math and the software developers tasked with implementing these techniques on large organizational datasets (as an example).  The point is that the commentary, parameter style, etc, all need to reflect a target audience that is often ignored when writing such scientific libraries.

We don't need an ELI5 series of comments or documentation efforts, but at least "Explain like I haven't thought about lingear algebra in a decade" is a fair target.  One way of achieving this isn't necessarily going into great detail on the _how_, but instead on the reasoning for _why_, as well as any interesting behaviors about how the range of values provided to a parameter may impact the resulting computation.

**Comment by dwaynepryce** [comment]
I should note that we should be doing both _how_ and _why_, not why over how.

A lot of software engineers need to understand your motivation, which we often glean from function or parameter names, commentary, etc.  But if you can construct a narrative around why you're doing the steps of your algorithm and how that achieves your goals, you can often make these seemingly opaque topics far more approachable and thus more used in consuming applications

**Comment by bdpedigo** [comment]
Separate thought: would be good to have a 1-sentence explanation of each of the build checks. I think this will help people who are new to the repo or CI in general to debug their own work with less intervention from maintainers

**Comment by bdpedigo** [comment]
addressed recently in #973
